dynamiqs/time_tensor.py

```diff
@@ -378,3 +424,91 @@ def __add__(self, other: Tensor | TimeTensor) -> TimeTensor:
             return PWCTimeTensor(factors, tensors, static=static)
         else:
             return NotImplemented
+
+
+class _ModulatedFactor:
+    # Defined by 2 tensors (factor, factor0), where
+    # - factor: (...,) is a callable that takes a time and returns a tensor
+    # - factor0: (...,) is the tensor returned by factor(0.0)
+    # factor0 holds information about the shape of the tensor returned by factor(t).
+
+    def __init__(self, factor: callable[[float], Tensor], factor0: Tensor):
+        self.factor = factor  # (...)
+        self.factor0 = factor0  # (...)
```

Comment:
Can we rename to `self.f` and `self.f0`? (to mimic `CallableTimeTensor`) Also in `totime()`